laborator/lab-07/README.md

```diff
@@ -0,0 +1,464 @@
+- Implementarea unei funcții în limbaj de asamblare
+- Folosirea stivei pentru a transmite parametrii unei funcții
+- Apelarea unei funcții externe (aflată în biblioteca standard C) din limbaj de asamblare
+
+## Transmiterea parametrilor
+ Cand vine vorba de a chema o functie cu parametri exista doua mari optiuni de plasare a acestora:
+
+1. Plasarea in registre - aceasta metoda, in mod intuitiv, presupune transmiterea parametrilor cu ajutorul registrelor.
+
+**Avantaje**
+
+- Este foarte usor de folosit atunci cand numarul parametrilor este mic.
+- Este foarte rapida, intrucat parametrii sunt imediat accesibili din registre.
+
+**Dezavantaje**
+
+- Din cauza faptului ca exista un numar limitat de registre, numarul de parametri ai unei functii ajunge sa fie limitat.
+- O alta problema o reprezinta faptul ca e foarte probabil ca unele registre sa fie folosite in interiorul functiei apelate. Prin urmare, e necesara salvarea temporara a registrelor inainte de a face apelul de functie. Ei bine, cel de-al doilea avantaj enumerat dispare, deoarece accesul la stiva presupune lucru cu memoria, adica latenta crescuta.
+
+2.Plasarea pe stiva - aceasta metoda presupune push-uirea pe stiva a tuturor parametrilor.
+
+**Avantaje**
+
+- Poate fi transmis un numar mare de parametri.
+
+**Dezavantaje**
+
+- Este lenta intrucat se face acces la memorie.
+- Mai complicata din punct de vedere al accesului la parametri.
+
+>**NOTE**:Pentru arhitecturiile **32-bit** se foloseste metoda plasarii pe stiva, iar pentru cele **64-bit** se foloseste metoda plasarii in registre. Noi vom folosi conventia de la **32-bit** architecture. 
+
+## Apelul unei funcții
+
+Atunci când apelăm o funcție, pașii sunt următorii:
+
+- Punem argumentele pe stivă, apelul de tip push fiind în ordinea   inversă în care sunt trimiși ca argumente funcției.
+- Apelăm call.
+- Restaurăm stiva la sfârșitul apelului.
+
+**Funcționarea stivei**
+
+După cum știm, operațiile pe stivă sunt de două tipuri:
+
+- **push val** în care valoarea **val** este plasată pe stivă
+- **pop reg/mem** în care ce se găsește în vârful stivei se plasează în registru sau într-o zonă de memorie
+
+În momentul în care se face **push** spunem că stiva **crește** (se adaugă elemente). În mod oarecum paradoxal însă, pointerul de stivă (indicat de registrul **esp** pe 32 de biți) scade. Acest lucru se întâmplă întrucât stiva crește în jos, de la adrese mari către adrese mici.
+
+La fel, în momentul în care facem **pop** spunem că stiva scade (se scot elemente). Acum pointer-ul de stivă (indicat de registrul **esp** pe 32 de biți) crește.
+
+Un sumar al acestui lucru este explicat foarte bine la acest link: [sumar](https://en.wikibooks.org/wiki/X86_Disassembly/The_Stack)
+
+Spre exemplu, daca avem functia foo cu urmatoarea semnatura (in limbaj C):
+
+```C
+int foo(int a, int b, int c);
+```
+
+Apelul acestei functii va arata astfel:
+
+```armasm
```

Comment:
Syntax highlight-ul din markdown-ul de pe GitHub este descris [aici](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks). Se folosește un utilitar numit linguist care acceptă diferite dialecte, descris la finalul paginii. Pentru assembly, ar trebui folosit [Assembly](https://github.com/github/linguist/blob/master/lib/linguist/languages.yml#L375), așa cum este specificat în fișierul yaml.